### PR TITLE
clang-format turicreate source code [1/2]

### PR DESCRIPTION
**Pull request by guihao-liang**
## Overview

write a program call `clang-format-all.py` to reformat all turicreate related cpp src into clang format.
```
(tc58) ➜  gui-1-36 git:(01-25-clang-format) ./scripts/clang-format-all.py -h
usage: clang-format-all.py [-h] [--path PATH] [--dry-run]
                           [--cfmt-args CFMT_ARGS [CFMT_ARGS ...]]
                           [--exclude-regex EXCLUDE_REGEX]
                           [--include-regex INCLUDE_REGEX]

format all CPP files recursively in a directory

optional arguments:
  -h, --help            show this help message and exit
  --path PATH           path to start the search
  --dry-run             list files wihtout invoking clang-format
  --cfmt-args CFMT_ARGS [CFMT_ARGS ...]
                        args for clang-format,  without any '-' in options
  --exclude-regex EXCLUDE_REGEX
                        files to exclude by regex pattern
  --include-regex INCLUDE_REGEX
                        files to exclude by regex pattern
```

## Sampe useage
```bash
./scripts/clang-format-all.py --path src/toolkits/drawing_classifier/ --cfmt-args style=google i
```

## Current Plan

- [ ] work on `toolkits` dir first to discover known limitations because it has some python docstring.
- [ ] work on rest of the CPP source files

## Known limitations

- [ ] model registrations have python docstrings that are treated as part of "CPP" source. I need manually reformat these files.

## Ultimate goal

Introducing clang-format pre-commit hooks

### CODE REVIEW

File: src/toolkits/activity_classification/ac_data_iterator.hpp [hunk 1]
```diff
@@ -213,34 +207,31 @@ class simple_data_iterator: public data_iterator {
...
- * \param[in] features              List of names of the columns containing the input features.
- * \param[in] session_id            Name of the column containing ids for each session in the dataset.
- *                                  A session is a single user time-series sequence.
- * \param[in] prediction_window     Number of time samples in every prediction window. A label is expected
- *                                  (for training), or predicted (in inference) every time a sequence of
- *                                  prediction_window samples have been collected.
- * \param[in] predictions_in_chunk  Each session is chunked into shorter sequences. This is the number of
- *                                  prediction windows desired in each chunk.
- * \param[in] target                Name of the coloumn containing the output labels. Empty string if None.
+ * \param[in] data                  Original data. Sframe containing one line
```
Comment (guihao-liang) [inline src/toolkits/activity_classification/ac_data_iterator.hpp]: need to resolve.

File: src/toolkits/activity_classification/activity_classifier.hpp [hunk 2]
```diff
@@ -125,17 +124,18 @@ class EXPORT activity_classifier: public ml_model_base {
       "----------\n"
       "data : SFrame\n"
       "    Dataset of new observations. Must include columns with the same\n"
-      "    names as the features used for model training, but does not require\n"
+      "    names as the features used for model training, but does not "
```
Comment (guihao-liang) [inline src/toolkits/activity_classification/activity_classifier.hpp]: need to resolve

**Comment by TobyRoseman** [comment]
@guihao-liang - it's not totally clear what you've written and what the tool has changed. I think it would be better to do one pull request for the tool you are building. Then another pull request for the changes the tool makes. I would recommend at least having these two things as separate commits, if not separate pull requests.

**Comment by guihao-liang** [comment]
> @guihao-liang - it's not totally clear what you've written and what the tool has changed. I think it would be better to do one pull request for the tool you are building. Then another pull request for the changes the tool makes. I would recommend at least having these two things as separate commits, if not separate pull requests.

Sure, I will split this pr into 2.

**Comment by guihao-liang** [comment]
close due to no further feedback.

**Pull request closed without merging**